tensorflow/python/keras/applications/imagenet_utils.py

```diff
@@ -97,9 +97,12 @@
 @keras_export('keras.applications.imagenet_utils.preprocess_input')
 def preprocess_input(x, data_format=None, mode='caffe'):
   """Preprocesses a tensor or Numpy array encoding a batch of images."""
+  if mode not in {'caffe', 'tf','torch'}:
+    raise ValueError('Unknown mode ' + str(mode))
```

Comment:
Can you add a test for this?

Reply:
Can you give me a little guidance as to what kind of test I should be writing? I'm new to contributing, a little pointer would help me a lot. Thank you!

Reply:
Please take a look at https://github.com/tensorflow/tensorflow/blob/aa070f5be6bfc55c25802540da149138696f965f/tensorflow/python/keras/applications/imagenet_utils_test.py#L29

We can add a simple test case for invalid mode here.

Reply:
Before I make a new PR for imagenet_utils_test.py, is this similar to what you had in mind?
```python
with self.assertRaises(ValueError):
    utils.preprocess_input(x, mode='some_unknown_mode')
```

Reply:
Yes

Reply:
Yes.